framework/include/base/FEProblem.h

```diff
@@ -481,12 +481,12 @@ class FEProblem :
    * @return Const reference to the user object
    */
   template <class T>
-  const T & getUserObject(const std::string & name)
+  const T & getUserObject(const std::string & name, unsigned int tid = 0)
```

Comment:
If you are really inside of the same threaded loop then this is valid code. However, you are modifying the FEProblem API. So this won't work in general. It won't even work if you try to retrieve an elemental user object from a side user object where we have a different threaded loop, or better yet an elemental pre-aux user object from an elemental post-aux user-object! We need to come up with a way of doing this without opening a hole in the framework that can be accidentally exploited.